File: app/src/main/kotlin/io/termplux/ui/view/ScrollControllerWebView.java
package io.termplux.ui.view;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.webkit.WebView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.kongzue.dialogx.interfaces.ScrollController;


public class ScrollControllerWebView extends WebView implements ScrollController {

    private boolean lockScroll;

    public ScrollControllerWebView(@NonNull Context context) {
        super(context);
    }

    public ScrollControllerWebView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public ScrollControllerWebView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public ScrollControllerWebView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    @Deprecated
    public boolean isLockScroll() {
        return lockScroll;
    }

    @Override
    public void lockScroll(boolean lockScroll) {
        this.lockScroll = lockScroll;
    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent event) {
        if (lockScroll) return false;
        return super.onTouchEvent(event);
    }

    @Override
    public int getScrollDistance() {
        return getScrollY();
    }

    @Override
    public boolean isCanScroll() {
        return true;
    }
}

File: android/app/src/main/kotlin/io/termplux/custom/ScrollControllerWebView.kt
package io.termplux.custom

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.webkit.WebView
import com.kongzue.dialogx.interfaces.ScrollController

class ScrollControllerWebView : WebView, ScrollController {

    constructor(
        context: Context
    ) : super(
        context
    )

    constructor(
        context: Context,
        attrs: AttributeSet?
    ) : super(
        context,
        attrs
    )

    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int
    ) : super(
        context,
        attrs,
        defStyleAttr
    )

    private var lockScroll = false

    override fun isLockScroll(): Boolean {
        return lockScroll
    }

    override fun lockScroll(lockScroll: Boolean) {
        this.lockScroll = lockScroll
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(event: MotionEvent): Boolean {
        return if (lockScroll) false else super.onTouchEvent(event)
    }

    override fun getScrollDistance(): Int {
        return scrollY
    }

    override fun isCanScroll(): Boolean {
        return true
    }
}